OpenRVS/classes/OpenServer.uc

```diff
@@ -20,7 +19,20 @@ event PreBeginPlay()
 	openrvs = none;//done initializing
 
 	super.PreBeginPlay();
-	SetTimer(60,true);
+	SetTimer(60, true);
+}
+
+//once everything is initialized, build a beacon text and automatically register
+//this server with the OpenRVS registry.
+event PostNetBeginPlay()
+{
+	local OpenBeacon beacon;
+
+	super.PostNetBeginPlay();
+
+	beacon = Spawn(class'OpenBeacon');

```

Comment:
There's no need to spawn a new beacon since every server has a beacon actor already.  You can grab a reference to it this way:
`Level.Game.m_UdpBeacon`
Also, technically since the beacon is an Actor, you would call Destroy() on it instead of just setting it to none.  Though if you use the server's existing beacon, please don't destroy it :)

Reply:
Hmm, I got this after making this change:
`OpenServer.uc(31) : Error, Unrecognized member 'm_UdpBeacon' in class 'GameInfo'`

Reply:
My bad, you need to cast to R6AbstractGameInfo.
`R6AbstractGameInfo(Level.Game).m_UdpBeacon`

Reply:
The class `R6AbstractGameInfo` adds `m_UdpBeacon` here: https://github.com/ijemafe/raven-shield-1.56/blob/19bed4d29d2f712df7b7ead226653712410204df/R6Abstract/Classes/R6AbstractGameInfo.uc#L41

Reply:
LOL I hadn't refreshed and we just posted the same class haha

Reply:
Ah, a followup:
`OpenServer.uc(32) : Error, Unrecognized member 'RegisterServer' in class 'UdpBeacon'`
The reference is to the base UdpBeacon rather than OpenBeacon.

Reply:
My bad again ... Need to cast from UdpBeacon to OpenBeacon.
`OpenBeacon(R6AbstractGameInfo(Level.Game).m_UdpBeacon)`

Reply:
Awesome! Thanks